### PR TITLE
expand crypto option

### PR DESCRIPTION
**Pull request by lukeburns**
Not sure this is of interest, but I wanted to try plugging in [`wasm-crypto`](https://github.com/jedisct1/wasm-crypto) in place of `hypercore-crypto` to use signatures over the Ristretto group. This PR makes it possible to specify a keyPair function that hypercore will use to generate keys, along with the size of keys/signatures. [lukeburns/wasm-crypto](https://github.com/lukeburns/wasm-crypto) has an api matching that of `hypercore-crypto` that can be used in the following example.

```js
var hypercore = require('hypercore')
var { keyPair, sign, verify } = require('wasm-crypto')

var crypto = {
  keyPair,
  sign (data, sk, cb) {
    return cb(null, sign(data, sk))
  },
  verify (data, sig, pk, cb) {
    return cb(null, verify(data, sig, pk))
  },
  secretKeySize: 32, // 32-byte secret keys
  publicKeySize: 32,
  signatureSize: 64
}

var feed = hypercore('./tmp', { valueEncoding: 'json', crypto })

feed.append({
  hello: 'world'
})

feed.append({
  hej: 'verden'
})

feed.append({
  hola: 'mundo'
})

feed.flush(function () {
  console.log('Appended 3 more blocks, %d in total (%d bytes)\n', feed.length, feed.byteLength)

  feed.createReadStream()
    .on('data', console.log)
    .on('end', console.log.bind(console, '\n(end)'))
})
```

### CODE REVIEW

File: lib/storage.js [hunk 1]
```diff
@@ -257,18 +265,16 @@ Storage.prototype.open = function (opts, cb) {
     })
   })
 
-  this.signatures.write(0, header(1, 64, 'Ed25519'), done)
+  this.signatures.write(0, header(1, this.signatureSize, 'Ed25519'), done)
```
Comment (emilbayes) [inline lib/storage.js]: Does the const string here also need changing? Ie. are the ristretto255 points "decompressed" back into valid ed25519 points?

File: index.js [hunk 2]
```diff
@@ -22,12 +22,13 @@ var safeBufferEquals = require('./lib/safe-buffer-equals')
...
 var defaultCrypto = {
+  keyPair: crypto.keyPair,
   sign (data, sk, cb) {
     return cb(null, crypto.sign(data, sk))
   },
 
-  verify (sig, data, pk, cb) {
-    return cb(null, crypto.verify(sig, data, pk))
+  verify (data, sig, pk, cb) {
+    return cb(null, crypto.verify(data, sig, pk))
```
Comment (emilbayes) [inline index.js]: Was the swapping of args here intentional?

Comment (lukeburns) [reply index.js]: Yes, to match the api of hypercore-crypto

File: lib/storage.js (hunk 1 above)
Comment (lukeburns) [reply lib/storage.js]: Ah yes we should change the string too. Does the protocol use this to ensure peers are using the same kind of signatures?

Comment (emilbayes) [reply lib/storage.js]: Not yet, but we’re working on updating the handshake to communicate info like this

Comment (lukeburns) [reply lib/storage.js]: Added signatureType

**Comment by mafintosh** [comment]
Closing cause old and lots have changed

**Pull request closed without merging**